// @nodefill/util/dist/types/index-node.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
const types_1 = __importDefault(require("node:util/types"));
exports.isExternal =
    exports.isDate =
        exports.isArgumentsObject =
            exports.isBigIntObject =
                exports.isBooleanObject =
                    exports.isNumberObject =
                        exports.isStringObject =
                            exports.isSymbolObject =
                                exports.isNativeError =
                                    exports.isRegExp =
                                        exports.isAsyncFunction =
                                            exports.isGeneratorFunction =
                                                exports.isGeneratorObject =
                                                    exports.isPromise =
                                                        exports.isMap =
                                                            exports.isSet =
                                                                exports.isMapIterator =
                                                                    exports.isSetIterator =
                                                                        exports.isWeakMap =
                                                                            exports.isWeakSet =
                                                                                exports.isArrayBuffer =
                                                                                    exports.isDataView =
                                                                                        exports.isSharedArrayBuffer =
                                                                                            exports.isProxy =
                                                                                                exports.isModuleNamespaceObject =
                                                                                                    exports.isAnyArrayBuffer =
                                                                                                        exports.isBoxedPrimitive =
                                                                                                            exports.isArrayBufferView =
                                                                                                                exports.isTypedArray =
                                                                                                                    exports.isUint8Array =
                                                                                                                        exports.isUint8ClampedArray =
                                                                                                                            exports.isUint16Array =
                                                                                                                                exports.isUint32Array =
                                                                                                                                    exports.isInt8Array =
                                                                                                                                        exports.isInt16Array =
                                                                                                                                            exports.isInt32Array =
                                                                                                                                                exports.isFloat32Array =
                                                                                                                                                    exports.isFloat64Array =
                                                                                                                                                        exports.isBigInt64Array =
                                                                                                                                                            exports.isBigUint64Array =
                                                                                                                                                                exports.isKeyObject =
                                                                                                                                                                    exports.isCryptoKey =
                                                                                                                                                                        void 0;
module.exports = types_1.default;
